backend/common/src/main/java/ai/verta/modeldb/common/futures/InternalFutureWithDefaultExecutor.java

```diff
@@ -0,0 +1,164 @@
+  }
+
+  @Override
+  @Deprecated
+  public <U> InternalFutureWithDefaultExecutor<U> thenApply(
+      Function<? super T, ? extends U> fn, Executor executor) {
+    return (InternalFutureWithDefaultExecutor<U>)
+        super.thenApply(fn, executor).thenWithExecutor(defaultExecutor);
+  }
+
+  public <U> InternalFutureWithDefaultExecutor<U> handle(
+      BiFunction<? super T, Throwable, ? extends U> fn) {
+    return handle(fn, defaultExecutor);
+  }
+
+  @Override
+  @Deprecated
+  public <U> InternalFutureWithDefaultExecutor<U> handle(
+      BiFunction<? super T, Throwable, ? extends U> fn, Executor executor) {
+    return (InternalFutureWithDefaultExecutor<U>)
+        super.handle(fn, executor).thenWithExecutor(defaultExecutor);
+  }
+
+  public <U, V> InternalFutureWithDefaultExecutor<V> thenCombine(
+      InternalFuture<? extends U> other, BiFunction<? super T, ? super U, ? extends V> fn) {
+    return thenCombine(other, fn, defaultExecutor);
+  }
+
+  @Override
+  @Deprecated
+  public <U, V> InternalFutureWithDefaultExecutor<V> thenCombine(
+      InternalFuture<? extends U> other,
+      BiFunction<? super T, ? super U, ? extends V> fn,
+      Executor executor) {
+    return (InternalFutureWithDefaultExecutor<V>)
+        super.thenCombine(other, fn, executor).thenWithExecutor(defaultExecutor);
+  }
+
+  public InternalFutureWithDefaultExecutor<Void> thenAccept(Consumer<? super T> action) {
+    return thenAccept(action, defaultExecutor);
+  }
+
+  @Override
+  @Deprecated
+  public InternalFutureWithDefaultExecutor<Void> thenAccept(
+      Consumer<? super T> action, Executor executor) {
+    return super.thenAccept(action, executor).thenWithExecutor(defaultExecutor);
+  }
+
+  public <U> InternalFutureWithDefaultExecutor<Void> thenRun(Runnable runnable) {
+    return thenRun(runnable, defaultExecutor);
+  }
+
+  @Override
+  @Deprecated
+  public <U> InternalFutureWithDefaultExecutor<Void> thenRun(Runnable runnable, Executor executor) {
+    return super.thenRun(runnable, executor).thenWithExecutor(defaultExecutor);
+  }
+
+  public InternalFutureWithDefaultExecutor<T> whenComplete(
+      BiConsumer<? super T, ? super Throwable> action) {
+    return whenComplete(action, defaultExecutor);
+  }
+
+  @Override
+  @Deprecated
+  public InternalFutureWithDefaultExecutor<T> whenComplete(
+      BiConsumer<? super T, ? super Throwable> action, Executor executor) {
+    return super.whenComplete(action, executor).thenWithExecutor(defaultExecutor);
+  }
+
+  public <U> InternalFutureWithDefaultExecutor<U> thenSupply(Supplier<InternalFuture<U>> supplier) {
+    return thenSupply(supplier, defaultExecutor);
+  }
+
+  @Override
+  @Deprecated
+  public <U> InternalFutureWithDefaultExecutor<U> thenSupply(
+      Supplier<InternalFuture<U>> supplier, Executor executor) {
+    return super.thenSupply(supplier, executor).thenWithExecutor(defaultExecutor);
+  }
+
+  public static class FactoryWithExecutor {
```

Comment:
i think I'd either call this simply `Factory` if we want to leave it as an inner class, or pull it up and call it `InternalFutureFactory`.